src/renderer/vt/XtermEngine.cpp

```diff
@@ -359,38 +375,28 @@ try
     const short dy = _scrollDelta.y<SHORT>();
     const short absDy = static_cast<short>(abs(dy));
 
-    HRESULT hr = S_OK;
-    if (dy < 0)
-    {
-        // Instead of deleting the first line (causing everything to move up)
-        // move to the bottom of the buffer, and newline.
-        //      That will cause everything to move up, by moving the viewport down.
-        // This will let remote conhosts scroll up to see history like normal.
-        const short bottom = _lastViewport.ToOrigin().BottomInclusive();
-        hr = _MoveCursor({ 0, bottom });
-        if (SUCCEEDED(hr))
-        {
-            std::string seq = std::string(absDy, '\n');
-            hr = _Write(seq);
-            // Mark that the bottom line is new, so we won't spend time with an
-            // ECH on it.
-            _newBottomLine = true;
-        }
-        // We don't need to _MoveCursor the cursor again, because it's still
-        //      at the bottom of the viewport.
-    }
-    else if (dy > 0)
+    // Shift our internal tracker of the last text position according to how
+    // much we've scrolled. If we manually scroll the buffer right now, by
+    // moving the cursor to the bottom row of the viewport and emitting a
+    // newline, we'll cause any wrapped lines to get broken.
+    //
+    // Instead, we'll just update our internal tracker of where the buffer
+    // contents are. On this frame, we'll then still move the cursor correctly
+    // relative to the new frame contents. To do this, we'll shift our
+    // coordinates we're tracking, like the row that we wrapped on and the
+    // position we think we left the cursor.
+    //
+    // See GH#5113
+    _lastText.Y += dy;

```

Comment:
with the newlining gone, how do we make sure that the pty host knows how many lines got pushed?

Reply:
explicit in the comment that was removed, too, is the pushing-into-scrollback behavior

Reply:
awit could you clarify? maybe it's just 5 o'clock calling, but I can't make sense of this comment 🙃

Reply:
Oh. This function used to print a set of newlines commensurate with the scroll distance, which would make the connected terminal push the scrolled-off lines into _its_ scrollback. The newlines are now dead, and I can't determine whether that broke anything.

Reply:
They seemingly _didn't_ break anything.

Conpty always emits a frame immediately before every new line, due to TriggerCircling. Basically we'll finish printing the frame before the newline, then we'll additionally paint another frame with `scrollDelta={0, -1}`, and in that second frame, since the cursor is now one line higher, we'll newline to move the cursor down, into the new line in the terminal.

Since we're emitting a frame _every_ newline, I'm not sure this function can actually get called with a delta != 1